Functions/Initialization/xASL_init_InitializeMutex.m

```diff
@@ -64,16 +64,80 @@
 %% --------------------------------------------------------
 %% 2) Initialize mutex object
 x.mutex = xASL_GoNoGo(x.dir.LockDir);
+
+%% 3) Check if this module is locked by another process
+bLocked = 0;
+
 if ~x.mutex.Lock(x.settings.MUTEXID)
-    fprintf('Also, check that there is no filesystem permission issue\n');
 	fprintf(2,['ERROR in module_' x.ModuleName ': mutex is locked: %s in %s\n'], x.settings.MUTEXID, x.dir.LockDir);
-	fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
+	fprintf('\n');
+    fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
 	fprintf('If this is not the case, the locked folder needs to be removed before proceeding\n');
 	fprintf('Also, check that there is no filesystem permission issue\n');
     fprintf('Otherwise this error can be ignored\n');
-	return;
-end
+    fprintf('\n');
+
+    bLocked = 1;
+else
+    
+    %% 4) Check if any module for this subject is locked
+    if ~strcmpi(x.ModuleName, 'population')
+        % the population module is subject-independent
+        if ~isfield(x, 'mutex') || ~isprop(x.mutex, 'Root')

```

Comment:
this all has to be done elsewhere.

Reply:
Per your other comments it seems that you agree now that we do this here right?
At the start of each module, `xASL_init_InitializeMutex` runs `xASL_GoNoGo`, which initializes `x.mutex.HasState` etc. `xASL_GoNoGo` also does a check if the current module is locked, but it doesn't check if other modules are locked (`xASL_GoNoGo` is essentially subject-agnostic, using its own module lock/status folder only)

I have now clarified this further in the comments

Reply:
Sounds good. Will check and test in detail.

Reply:
This works really well now. I mean the concept is nice, but there are some bugs :)
You can resolve this issue, but keep the others open for now.